internal/status.go

```diff
@@ -0,0 +1,147 @@
+package internal
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/dustin/go-humanize"
+)
+
+type StatusLine struct {
+	resources []Resource
+
+	resourceSizes       []int64
+	totalBytes          int64
+	sizingSuccess       bool
+	bytesDownloaded     int64
+	resourcesDownloaded int
+	spinI               int
+	indexCh             chan int
+	startTime           time.Time
+}
+
+var spinChars = [5]string{"-", "\\", "|", "/", "-"}
+
+// Create, initialize, and start a new StatusLine.
```

Comment:
See the doc about how inline doc is formatted: https://tip.golang.org/doc/comment#func
It's more idiomatic to start these comments with the name of the function. Like: `NewStatusLine creates, initializes, and starts a new StatusLine`.